### ISSUE TITLE
Quick Fix `Add "Optional" to Type Annotation` doesn't work

### ISSUE THREAD

**Issue by Molkree** [microsoft/pylance-release#1873]
## Environment data

-   Language Server version: v2021.9.3
-   OS and version: Win10, 19043.1237
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.7

## Expected behaviour

Quick Fix wraps function parameter type in `Optional`.

## Actual behaviour

This is with strict type checking. `None` is underlined with this error:
```
Expression of type "None" cannot be assigned to parameter of type "int"
  Type "None" cannot be assigned to type "int" Pylance(reportGeneralTypeIssues)
```
When trying to apply Quick Fix nothing happens.

## Code Snippet / Additional information

```python
def foo(arg: int = None):
    print(arg)
```

**Comment by jakebailey** [comment]
I debugged this, and this one's just a bug where we seemingly have forgotten to include the name of the file to edit on the quick fix.

The fix is simple, but will take a little more than a line thanks to some plumbing to get a URI to the right place.

**Comment by CodeCrazy-ywt** [comment]
But what about an argunent that starts life as None and stays not-None once assigned not-None object?

**Comment by erictraut** [comment]
I'm not sure what you mean. Can you provide a concrete example? I presume you're talking about a parameter, not an argument? (An argument is a value passed to a function, whereas a parameter is the name of an input used within a function.)

**Comment by CodeCrazy-ywt** [comment]
Here's an example:
```python
def func(lst: list[int], start: int = 0, end: int = None):
    if end is None:
        end = len(lst)
    # end is no longer None
    ...
```

**Comment by jakebailey** [comment]
I don't really know what the above has to do with the code action being broken; if you don't want to apply the code action, you don't have to. But, in the case above, if you type `end` as `Optional[int]`, the code that assigns over `end` with an `int` if it's None will mean further uses are narrowed to `int` anyway.

**Comment by CodeCrazy-ywt** [comment]
Ok, thank u, I understand.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202194-29-september-2021
